src/AdminConsole/Pages/Organization/Admins.cshtml.cs

```diff
@@ -92,28 +94,39 @@ public async Task<IActionResult> OnPostDelete(string userId)
     public async Task<IActionResult> OnPostInvite(InviteForm form)
     {
         CanInviteAdmin = await _dataService.CanInviteAdminAsync();
+
         if (CanInviteAdmin is false)
         {
             ModelState.AddModelError("error", "You need to upgrade to a paid organization to invite more admins.");
             return await OnGet();
         }
 
+        var org = await _dataService.GetOrganizationAsync();
+        var existingInvites = await _invitationService.GetInvitesAsync(org.Id);
+
+        if (existingInvites.Count >= 10)
+        {
+            ModelState.AddModelError("error", "You can only have 10 pending invites at a time.");
+            return await OnGet();
+        }
+
+        if (existingInvites.Any(x => string.Equals(x.ToEmail, form.Email, StringComparison.CurrentCultureIgnoreCase)))
+        {
+            ModelState.AddModelError("error", "There is a pending invite already for this address. Please cancel before resending.");
+            return await OnGet();
+        }
+
         if (!ModelState.IsValid)
         {
             // todo: Is there a pattern where we don't need to repeat this?
             return await OnGet();
         }
 
-        Models.Organization org = await _dataService.GetOrganizationAsync();
-        var orgId = org.Id;
-        var orgName = org.Name;
-        ConsoleAdmin user = await _dataService.GetUserAsync();
-        var userEmail = user.Email;
-        var userName = user.Name;
+        var user = await _dataService.GetUserAsync();
 
-        await _invitationService.SendInviteAsync(form.Email, orgId, orgName, userEmail, userName);
+        await _invitationService.SendInviteAsync(form.Email, org.Id, org.Name, user.Email!, user.Name);
 
-        _eventLogger.LogInviteAdminEvent(user, form.Email, _systemClock.UtcNow.UtcDateTime);
+        _eventLogger.LogInviteAdminEvent(user, form.Email, _timeProvider.GetUtcNow().UtcDateTime);
```

Comment:
Nit/future: Maybe `TimeProvider` can be injected in some kind of context inside the event logger so that `_timeProvider.GetUtcNow().UtcDateTime` can be odne implicitly?